GameClasses.ts

```diff
@@ -0,0 +1,89 @@
+import {Card, DecreaseCard, EffectCard, IncreaseCard} from "./CardsClasses";
+
+class Points {
+    private score: number = 0
+    private points: number = this.getPoints(6, 10)
+
+    increase() {
+        this.score += this.points
+    }
+    decrease() {
+        this.score -= this.points
+    }
+
+    getPoints(min: number, max: number): number {
+        return Math.floor(Math.random() * (max - min) + min)
+    }
+
+    get currentScore() {
+        return this.score
+    }
+    get currentPoints() {
+        return this.points
+    }
+}
+
+class Game {
+    private cards: Card[]
+
+    constructor() {
+        this.cards = this.getDeck(6, 10)
+    }
+
+    public run(symbol: string) {
+        if (this.cards) {
+            console.log(this.cards)
+        }
+
+        symbol = symbol.toUpperCase()
+        let activeCads: Card[] = []
+        this.cards = this.cards.filter((el: Card) => {
+            if (el.symbol !== symbol) {
+                return el
+            } else {
+                activeCads.push(el)
+            }
+        })
+
+        if (activeCads) {
+            activeCads.forEach((el) => {
+                el.effect()
+            })
+        }
+
+        console.log('----------')
+        return `TOTAL SCORE: ${score.currentScore}`
+    }
+
+    public getRandomSymbol(): string {
+        enum Symbols {
+            'A' = 0,
+            'B' = 1,
+            'C' = 2
+        }
+
+        return Symbols[Math.floor(Math.random() * 3)]
+    }
+
+    private getRandomCard(): Card {
+        const cards: Card[] = [
```

Comment:
As far as I understand you use only one Card from this list? What if you have 10 types of Cards or more? You'll create 10 Card to use the only one. Let's try to find a more elegant solution during design patterns practice!

Reply:
Will make new issue for this comment

Reply:
Could you post a link here?

Reply:
@silchencko we decided to put it off for a while

Reply:
@silchencko This is the [link](https://github.com/what-the-hub/ts-card-game/issues/2) for this issue